src/client/javascripts/cookie-consent.test.js

```diff
@@ -0,0 +1,242 @@
+import { beforeEach, describe, expect, test, vi } from 'vitest'
+
+const createMockElement = (type = 'div') => ({
+  setAttribute: vi.fn(),
+  removeAttribute: vi.fn(),
+  getAttribute: vi.fn(),
+  appendChild: vi.fn(),
+  addEventListener: vi.fn(),
+  insertBefore: vi.fn(),
+  tagName: type.toUpperCase(),
+  textContent: '',
+  src: '',
+  height: '',
+  width: '',
+  style: {
+    display: '',
+    visibility: ''
+  },
+  dataset: {}
+})
+
+const mockDocument = {
+  cookie: '',
+  readyState: 'complete',
+  getElementById: vi.fn(),
+  createElement: vi.fn((type) => createMockElement(type)),
+  querySelector: vi.fn(),
+  head: { appendChild: vi.fn() },
+  body: { insertBefore: vi.fn(), firstChild: createMockElement() },
+  addEventListener: vi.fn()
+}
+
+globalThis.document = mockDocument
+
+const DEFAULT_BANNER_CONFIG = {
+  cookieName: 'cookie_consent',
+  expiryDays: '365',
+  gaTrackingId: 'GA-123'
```

Comment:
whats happens if this is undefined? like in an environment where we don't have GA set